tree/30_generic_methods/file_copy_from_local_source.cf

```diff
@@ -28,6 +28,11 @@
 
 bundle agent file_copy_from_local_source(source, destination)
 {
+  vars:
+     "class_prefix"        string => canonify("file_copy_from_local_source_${destination}");
+     "new_class_prefix"    string => canonify(join("_", "this.callers_promisers"));
+
   methods:
-      "placeholder" usebundle => file_copy_from_local_source_recursion("${source}", "${destination}", "0");
+      "copy without recursion" usebundle => file_copy_from_local_source_recursion("${source}", "${destination}", "0");
+      "new result classes"     usebundle => _classes_copy("${class_prefix}", "${new_class_prefix}");
```

Comment:
The second time you call the file_copy_from_local_source method, you will have a different promiser, which will give a different stack from here, hence a different new_class_prefix.
We will still have a common class_prefix, but we we cannot get rid of it until we are sure that the promiser stack feature is supported.